src/Core/Models/Business/SecretsManagerSubscriptionUpdate.cs

```diff
@@ -8,87 +9,95 @@ public class SecretsManagerSubscriptionUpdate
     public Organization Organization { get; set; }
 
     /// <summary>
-    /// The seats to be added or removed from the organization
+    /// The total seats the organization will have after the update, including any base seats included in the plan
     /// </summary>
-    public int SmSeatsAdjustment { get; set; }
+    public int? SmSeats { get; set; }
 
     /// <summary>
-    /// The total seats the organization will have after the update, including any base seats included in the plan
+    /// The new autoscale limit for seats, expressed as a total (not an adjustment).
+    /// This may or may not be the same as the current autoscale limit.
     /// </summary>
-    public int SmSeats { get; set; }
+    public int? MaxAutoscaleSmSeats { get; set; }
 
     /// <summary>
-    /// The seats the organization will have after the update, excluding the base seats included in the plan
-    /// Usually this is what the organization is billed for
+    /// The total service accounts the organization will have after the update, including the base service accounts
+    /// included in the plan
     /// </summary>
-    public int SmSeatsExcludingBase { get; set; }
+    public int? SmServiceAccounts { get; set; }
 
     /// <summary>
-    /// The new autoscale limit for seats, expressed as a total (not an adjustment).
+    /// The new autoscale limit for service accounts, expressed as a total (not an adjustment).
     /// This may or may not be the same as the current autoscale limit.
     /// </summary>
-    public int? MaxAutoscaleSmSeats { get; set; }
+    public int? MaxAutoscaleSmServiceAccounts { get; set; }
 
     /// <summary>
-    /// The service accounts to be added or removed from the organization
+    /// The proration date for the subscription update (optional)
     /// </summary>
-    public int SmServiceAccountsAdjustment { get; set; }
+    public DateTime? ProrationDate { get; set; }
 
     /// <summary>
-    /// The total service accounts the organization will have after the update, including the base service accounts
-    /// included in the plan
+    /// Whether the subscription update is a result of autoscaling
     /// </summary>
-    public int SmServiceAccounts { get; set; }
+    public bool Autoscaling { get; init; }
```

Comment:
@r-tome I'm undecided whether the Autoscaling property is a good idea. It complicates the logic a bit, and it's only used (mostly) to show more accurate error messages. Could we just make the error messages more general and then remove this property? I might experiment on Monday but wanted your thoughts.

Reply:
I doesn't look terrible but if we can simplify it then I think we should.

One possibility on how we could get around this would be have all validation methods moved to `SecretsManagerSubscriptionUpdate` and then create another class that inherits from that (say `SecretsManagerSubscriptionAutoscale`) and override the validation methods with autoscale logic and appropriate messages. I'm not sure if that would work and maybe it's overcomplicating it. We can sync up sometime and discuss it.

Reply:
I have considered moving the validation checks inside the object, I'm undecided on it. It might be better - you could have an `OrganizationSubscriptionView` in the db which you query, that returns everything you need including the current usage etc, then `Validate` just compares the internal properties of the object.

I don't think that it helps with this particular question though, one thing I want to ensure is that anyone calling this does not accidentally skip autoscale validation checks. The nice thing about the constructor signature at the moment is that you _have_ to specify whether you're autoscaling or not. Compare to needing to instantiate a subclass, or call a different method - both would be easy to miss.

I reviewed today and it's not as bad as I remember from Friday. I think we can leave as is.